pkg/controller/sidecarset/sidecarset_controller.go

```diff
@@ -131,7 +130,29 @@ func (r *ReconcileSidecarSet) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
-	// update sidecarset status
 	err = r.updateSidecarSetStatus(sidecarSet, status)
-	return reconcile.Result{}, err
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	unavailableNum, err := getUnavailableNumber(sidecarSet, filteredPods)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	// we only support sequential update currently, equals to maxUnavailable = 1
+	if unavailableNum != 0 {
+		klog.V(3).Infof("current unavailable pod number: %v, skip update", unavailableNum)
+		return reconcile.Result{}, nil
+	}
+
+	if sidecarSet.Spec.Paused {
```

Comment:
Should we move Pause checking before unavailable pods checking ? after all, if sidecarset is paused, unavailable number is no longer used.

Reply:
done